### DISCUSSION TITLE
JBD UP16s010

### DISCUSSION THREAD

**Discussion by Herculezz55** [syssi/esphome-jbd-bms#74] [General]
Anyone tried this with this model?

**Comment by jasonxcool** [comment]
I have "JBD UP16S015" BMS running!

**Reply by syssi** [reply]
@jasonxcool Could you provide some details about your BMS? I would like to add your modele to the list of supported devices.

**Reply by Ihorvovchok** [reply]
> I have "JBD UP16S015" BMS running!

Could you please share your Home assistant config for UP16S015. Thank you.

**Comment by jasonxcool** [comment]
Meinen Code für ESPHome findet ihr Hier:
[https://github.com/syssi/esphome-jbd-bms/discussions/106](url)

Im  Home Assistant habe ich keine änderung an den yaml vorgenommen, wurde als Gerät mit entsprechenden Entitäten erkannt

Nähere Angaben zum BMS habe ich leider nicht. Da ich scheinbar damals ein Prototypen bekommen habe.
Seriennummer: 001

**Reply by Ihorvovchok** [reply]
Thank you. Everything worked out for me.

**Comment by stefan064** [comment]
The UP16S010 uses a slightly different communication protocol to other JBD BMS - they added an 'address' field and some other minor changes that render this model unfortunately incompatible with the usual JBD protocol here.
Most notable is this variant allows multiple BMS to talk over a single multidrop RS485 bus to a single ESP.
Another user kindly implemented this in a fork, which I've forked again to add what I needed:
https://github.com/stefan064/esphome-jbd-bms
It would be great to add this variant as an option here.
